selftests/unit/test_job.py

```diff
@@ -171,6 +171,13 @@ def test_job_no_base_logdir(self):
         self.assertEqual(os.path.dirname(empty_job.logdir), self.tmpdir)
         self.assertTrue(os.path.isfile(os.path.join(empty_job.logdir, 'id')))
 
+    def test_job_dryrun_no_base_logdir(self):
+        args = argparse.Namespace(dry_run=True)
+        empty_job = job.Job(args)
+        self.assertTrue(os.path.isdir(empty_job.logdir))
+        self.assertTrue(os.path.isfile(os.path.join(empty_job.logdir, 'id')))
+        shutil.rmtree(empty_job.args.base_logdir)
```

Comment:
This should be in `try/finally` block (after you get the `empty_job`), or ideally you should patch it to provide your tmpdir to avoid files left behind in case the assertion fails.

Reply:
True, adding a try/except block.